### ISSUE TITLE
Linux Support

### ISSUE THREAD

**Issue by vrothberg** [redhat-developer/podman-desktop-redhat-account-ext#71]
I want to kick off a conversation about how to best support Linux.  My current understanding is as follows (please correct if I am off):

- podman-machine is not fully supported on Linux as qemu isn't packaged along with Podman Desktop.
- A podman-machine cannot be created via Podman Desktop on Linux.
- The extension - at the present - requires a podman-machine running and will otherwise error out.
- subscription-manager is only supported on distributions in the RH ecosystem (Fedora, CentOS, RHEL).

On Fedora, CentOS and RHEL we could run subscription-manager directly on the host.  Installing sub-man and running it requires root privileges, so we'd need a sudo prompt (e.g., https://www.npmjs.com/package/sudo-prompt) to run the installation and registration in a privileged process.

I am personally OK to keep the requirement of having a podman-machine running as on Mac OS and Windows BUT the support for that is very limited on Linux.  Using the host, in turn, is limited to distribution shipping sub-man.  So I am currently gravitating toward a better podman-machine support on Linux.

Cc: @slemeur @stefwalter @benoitf

**Comment by slemeur** [comment]
My early sentiment on this, is that Podman on linux should run by default without the need of a VM (and podman-machine).

As soon as the subscription-manager would be required, it might be the time where we transition the environment, or where we started an alternate podman-machine for particular tasks - but that would bring some UX challenges (and potentially performances ones too).

**Comment by odockal** [comment]
I was playing around on linux today with SSO and Podman Machine. There is an issue worth considering if one is using podman machine while trying to get the auth. token from SSO login. 
basically if you are using podman machine (it is started from cli) and tries to log in, the request is not finished, here is the error:
```
main ↪️ (node:3) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/run/user/1000/containers/auth.json'

main ↪️ (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 15)
console.<computed> @ VM5:2
(anonymous) @ VM5:2
emit @ VM4 sandbox_bundle:2
onMessage @ VM4 sandbox_bundle:2

main ↪️ [redhat-authentication] Subscription manager registration returned exit code: 125
console.<computed> @ VM5:2
(anonymous) @ VM5:2
emit @ VM4 sandbox_bundle:2
onMessage @ VM4 sandbox_bundle:2

```
And

**Comment by artificial-intelligence** [comment]
any update on this? see also my recent bug report:

https://github.com/redhat-developer/podman-desktop-redhat-account-ext/issues/574

and this podman-desktop discussion I kicked off:

https://github.com/containers/podman/discussions/25308

**Comment by vrothberg** [comment]
I'd love the requirement on podman machine on Linux to be lifted.  The bootc extension already supports running rootful podman if needed (e.g., for converting a bootc to a disk image).

<img width="1728" alt="Image" src="https://github.com/user-attachments/assets/7074c11f-94db-4bdd-8d2f-a774a34a6180" />

I believe the RH account extension can follow the example of the bootc extension and prompt the user for credentials when needed (installing/running subscription-manager):

<img width="1728" alt="Image" src="https://github.com/user-attachments/assets/d25d55fd-da73-4a3b-b55d-625d92933da9" />

**Comment by vrothberg** [comment]
@deboer-tim @slemeur @mrguitar FYI

**Comment by mrguitar** [comment]
I can see three scenarios for this.

1) Host is already subscribed w/ subscription-manager - no podman machine: This is the scenario we'll have most often with RHEL and the extension really only handles the registry auth. I think we're good here, this is what I use on my daily driver and it's awesome.
2) Host is not subscribed (sub-man may not be installed) - no podman machine: This is where we have a gap.
3) podman machine is in use: It's been a few months since I've done this on Linux, but I believe this scenario works well like on Mac & Win.

All this to say that the 1st scenario is where we'll be on the RHEL side, and I think it's valid to ask fedora & centos-stream users to register with subscription-manager. Thoughts?

**Comment by vrothberg** [comment]
>     1. Host is already subscribed w/ subscription-manager - no podman machine: This is the scenario we'll have most often with RHEL and the extension really only handles the registry auth. I think we're good here, this is what I use on my daily driver and it's awesome.

For the record (as @mrguitar and I just ran through it).  In this case, the extension will still display an error BUT the containers will work just fine.

>     2. Host is not subscribed (sub-man may not be installed) - no podman machine: This is where we have a gap.

Yes, that's something I'd love to solve such that the extension prompts the user for credentials, installs the sub-man and registers the host.
 
>     3. podman machine is in use: It's been a few months since I've done this on Linux, but I believe this scenario works well like on Mac & Win.

Yes, this will work just like on Mac & Win.

**Comment by mrguitar** [comment]
Yup, I'm an idiot and forgot or didn't realize I'm getting the same error. Apparently I dismiss those with the same reckless abandonment as legal terms & services. :(

**Comment by cdrage** [comment]
> I'd love the requirement on podman machine on Linux to be lifted. The bootc extension already supports running rootful podman if needed (e.g., for converting a bootc to a disk image).
> 
> <img alt="Image" width="1728" src="https://private-user-images.githubusercontent.com/11679875/439169221-7074c11f-94db-4bdd-8d2f-a774a34a6180.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYwMjMyMzgsIm5iZiI6MTc0NjAyMjkzOCwicGF0aCI6Ii8xMTY3OTg3NS80MzkxNjkyMjEtNzA3NGMxMWYtOTRkYi00YmRkLThkMmYtYTc3NGEzNGE2MTgwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDMwVDE0MjIxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExMzZmODljYTY5NDVmMjNlNWFjYzM0MjVlMzkyY2E4ZTJmZTQ3YmZjNjdjODc3NzYyM2RlZGY4OTg1ZTQ1OTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gVX4ANisQn1R-GinjvPUU1R87Pqe4eGHSA-S4d78icA">
> I believe the RH account extension can follow the example of the bootc extension and prompt the user for credentials when needed (installing/running subscription-manager):
> 
> <img alt="Image" width="1728" src="https://private-user-images.githubusercontent.com/11679875/439171409-d25d55fd-da73-4a3b-b55d-625d92933da9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYwMjMyMzgsIm5iZiI6MTc0NjAyMjkzOCwicGF0aCI6Ii8xMTY3OTg3NS80MzkxNzE0MDktZDI1ZDU1ZmQtZGE3My00YTNiLWI1NWQtNjI1ZDkyOTMzZGE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDMwVDE0MjIxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE3MGVkOTBhNWFlMWY5YTlkMzhhNmM3NjY1Nzg5ZjI1MjZiNzc0MzdlZTAxMmM1YTdjMDBkODdiYzVmZDI2NGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8BluLavAPn4m-hZgw7PGTY59U7t4g3Xq-p2Ybrl2U2E">

Oh FYI! We do not use Podman Machine on Linux for the bootc extension.

Last year we moved away from podman machine since it was causing multiple issues and now you can build it natively (we use process.exec under the hood).

**Comment by vrothberg** [comment]
Thanks, @cdrage !  That's exactly what I meant (but probably wasn't clear about it).  I want the account extension to behave like the bootc one in this regard.
